src/libasr/codegen/asr_to_python.cpp

```diff
@@ -573,6 +573,58 @@ class ASRToLpythonVisitor : public ASR::BaseVisitor<ASRToLpythonVisitor>
         s += "len(" + s + ")";
     }
 
+    void visit_StringItem(const ASR::StringItem_t &x) {
+        std::string r = "";
+        visit_expr(*x.m_arg);
+        r += s;
+        r += "[";
+        visit_expr(*x.m_idx);
+        r += s;
+        r += "]";
+        s = r;
+    }
+
+    void visit_StringSection(const ASR::StringSection_t &x) {
+        std::string r = "";
+        visit_expr(*x.m_arg);
+        r += s;
+        if (x.m_start) {
+            r += "[";
+            visit_expr(*x.m_start);
+            r += s;
+        }
+        if (x.m_end) {
+            r += ":";
+            visit_expr(*x.m_end);
+            r += s;
+        }
+        if (x.m_step) {
+            r += ":";
+            visit_expr(*x.m_step);
+            r += s;
+            r += "]";
+        } else {
+            r += "]";
+        }
```

Comment:
```suggestion
        r += "[";
        if (x.m_start) {
            visit_expr(*x.m_start);
            r += s;
        }
        r += ":";
        if (x.m_end) {
            visit_expr(*x.m_end);
            r += s;
        }
        if (x.m_step) {
            r += ":";
            visit_expr(*x.m_step);
            r += s;
        }
        r += "]";
```

I suggested this changes because of the following cases:
```
s: str = "12345"
print(s[:])
print(s[3:])
print(s[3:4:])
```

Reply:
Leave comment for proper understanding  of code.